src/main/java/com/bina/lrsim/SamplerDriver.java

```diff
@@ -76,17 +76,23 @@ else if(this.reference.exists()) {
         }
         return null;
       } else if (readType.equals("bax")) {
-        if (this.inFile.endsWith("cmp.h5")) {
+        if (this.inFile.endsWith(".cmp.h5")) {
```

Comment:
We should have these suffix names as constants in some file. HTSJDK has also defined SAM/BAM/CRAM constants., e.g., http://static.javadoc.io/com.github.samtools/htsjdk/1.132/htsjdk/samtools/BamFileIoUtils.html#BAM_FILE_EXTENSION